modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -265,283 +365,280 @@ FeatureFloodCount::getElementalValues(dof_id_type /*elem_id*/) const
-
-      // Now pack it up
-      unsigned int current_idx = 0;
-
-      // Note: The zeroth "region" is everything outside of a bubble - we don't want to put
-      // that into our packed data structure so start at 1 here!
-      for (unsigned int i = 0; i < _region_counts[map_num]; ++i)
+    for (processor_id_type rank = 0; rank < n_procs; ++rank)
+      for (std::vector<FeatureData>::iterator it = _partial_feature_sets[rank][map_num].begin();
+           it != _partial_feature_sets[rank][map_num].end(); ++it)
       {
-        partial_packed_data[current_idx++] = data[map_num][i].size();     // The number of nodes in the current region
+        FeatureData & feature = *it;
 
-        if (_single_map_mode)
+        for (std::set<dof_id_type>::iterator entity_it = feature._local_ids.begin(); entity_it != feature._local_ids.end(); ++entity_it)
         {
-          mooseAssert(i < _region_to_var_idx.size(), "Index out of bounds in FeatureFloodCounter");
-          partial_packed_data[current_idx++] = _region_to_var_idx[i];     // The variable owning this bubble
+          dof_id_type entity_id = *entity_it;
+
+          // TODO: This may not be good enough for the elemental case
+          const Point & entity_point = _is_elemental ? mesh.elem(entity_id)->centroid() : mesh.node(entity_id);
+
+          /**
+           * Update the bounding box.
+           *
+           * Note: There will always be one and only one bbox while we are building up our
+           * data structures because we haven't started to stitch together any regions yet.
+           */
+          feature.updateBBoxMin(feature._bboxes[0], entity_point);
+          feature.updateBBoxMax(feature._bboxes[0], entity_point);
+
+          // Save off the min entity id present in the feature to uniquely identify the feature regardless of n_procs
+          feature._min_entity_id = std::min(feature._min_entity_id, entity_id);
         }
-        else
-          partial_packed_data[current_idx++] = map_num;                   // The variable owning this bubble
 
-        std::set<dof_id_type>::iterator end = data[map_num][i].end();
-        for (std::set<dof_id_type>::iterator it = data[map_num][i].begin(); it != end; ++it)
-          partial_packed_data[current_idx++] = *it;                       // The individual entity ids
+        // Adjust the halo marking region
+        std::set<dof_id_type> set_difference;
+
+        std::set_difference(feature._halo_ids.begin(), feature._halo_ids.end(), feature._local_ids.begin(), feature._local_ids.end(),
+                            std::insert_iterator<std::set<dof_id_type> >(set_difference, set_difference.begin()));
+        feature._halo_ids.swap(set_difference);
+
+        // Periodic node ids
+        appendPeriodicNeighborNodes(feature);
       }
 
-      packed_data.insert(packed_data.end(), partial_packed_data.begin(), partial_packed_data.end());
-    }
-  }
+//  // DEBUGGING
```

Comment:
This commented out "DEBUGGING" block should be removed.  You can do it in a separate commit though, so you can always get it back easily later.